grammars/clojure.cson

```diff
@@ -13,6 +13,25 @@
 ]
 'foldingStartMarker': '\\(\\s*$'
 'foldingStopMarker': '^\\s*\\)'
+'firstLineMatch': '''(?x)
+  # Hashbang
+  ^\\#!.*(?:\\s|\\/)
+    boot
+  (?:$|\\s)
+  |
+  # Modeline
+  (?i:
+    # Emacs
+    -\\*-(?:\\s*(?=[^:;\\s]+\\s*-\\*-)|(?:.*?[;\\s]|(?<=-\\*-))mode\\s*:\\s*)
+      (?:clojure|clojurescript)
```

Comment:
I haven't been nitpicking about this in the other PRs because it really doesn't matter, but I don't believe the firstLineMatch cares about captures, so you shouldn't need to mark all the capture groups as non-capturing.

So then this line can be simplified to `(clojure(script)?)`.